mongosql/src/algebrizer/test.rs

```diff
@@ -7454,20 +7454,256 @@ mod limit_or_offset_clause {
-        expected_error_code = 3006,
+        expected = Ok(mir::Stage::Project(mir::Project {
+            source: Box::new(mir::Stage::Group(mir::Group {
+                source: Box::new(mir::Stage::Set(mir::Set {
+                    operation: mir::SetOperation::UnionAll,
+                    left: Box::new(mir::Stage::Project(mir::Project {
+                        source: Box::new(mir::Stage::Collection(mir::Collection {
+                            db: "test".into(),
+                            collection: "foo".into(),
+                            cache: SchemaCache::new()
+                        })),
+                        expression: map! {
+                            (DatasourceName::Named("foo".into()), 0u16).into() => mir::Expression::Reference((DatasourceName::Named("foo".into()), 0u16).into())
+                        },
+                        is_add_fields: false,
+                        cache: SchemaCache::new()
+                    })),
+                    right: Box::new(mir::Stage::Project(mir::Project {
+                        source: Box::new(mir::Stage::Collection(mir::Collection {
+                            db: "test".into(),
+                            collection: "bar".into(),
+                            cache: SchemaCache::new()
+                        })),
+                        expression: map! {
+                            (DatasourceName::Named("bar".into()), 0u16).into() => mir::Expression::Reference((DatasourceName::Named("bar".into()), 0u16).into())
+                        },
+                        is_add_fields: false,
+                        cache: SchemaCache::new()
+                    })),
+                    cache: SchemaCache::new()
+                })),
+                keys: vec![
+                    mir::OptionallyAliasedExpr::Aliased(mir::AliasedExpr {
+                        alias: "__groupKey0".to_string(),
+                        expr: mir::Expression::Reference(
+                            (DatasourceName::Named("bar".to_string()), 0u16).into()
+                        )
+                    }),
+                    mir::OptionallyAliasedExpr::Aliased(mir::AliasedExpr {
+                        alias: "__groupKey1".to_string(),
+                        expr: mir::Expression::Reference(
+                            (DatasourceName::Named("foo".to_string()), 0u16).into()
+                        )
+                    })
+                ],
+                aggregations: vec![],
+                cache: SchemaCache::new(),
+                scope: 0
+            })),
+            expression: map! {
+                (DatasourceName::Named("bar".to_string()), 0u16).into() => mir::Expression::FieldAccess(mir::FieldAccess {
+                    expr: Box::new(mir::Expression::Reference((DatasourceName::Bottom, 0u16).into())),
+                    field: "__groupKey0".to_string(),
+                    is_nullable: true
+                }),
```

Comment:
```suggestion
                ("bar", 0u16).into() => crate::util::mir_field_access("__bot__", "__groupKey0", true),
```
We have some more test helpers in `crate/util/mod.rs` that could again help reduce test size.

Reply:
I think a good skunkworks project would be adding serde for mir so that we can make these tests less hideous :joy:

Reply:
That would be wonderful 🙏